crates/base_db/src/input.rs

```diff
@@ -192,6 +192,7 @@ pub struct CrateData {
     pub env: Env,
     pub dependencies: Vec<Dependency>,
     pub proc_macro: Vec<ProcMacro>,
+    pub features: FxHashMap<String, Vec<String>>,
```

Comment:
We shouldn't be adding `features` concept at this level. Feature are Cargo-specific, but this code is build-system independent. 

See https://github.com/rust-analyzer/rust-analyzer/pull/9081/files#diff-40bbf5978d3aa37e21254b0214471eb1a6f1dcee7f0d8f6ce7d10f855f2a2cf7L147-R151

What you want to use instead is `cfg_options`.

Reply:
I did look at `cfg_options` as I was using them for the other part of completion, and it wasn't clear to me if they should only be enabled options though, so is it ok to add values to `cfg_options` that are currently disabled?

Reply:
Yeah, `cfg_options` list only enabled options. It would be fine to add 

```rust
  potential_cfg_options: CfgOptions
```

which are the super-set of `cfg_options` and specify what else can be enabled in theory.